### DISCUSSION TITLE
lua-cjson项目decode null值会返回userdata

### DISCUSSION THREAD

**Discussion by ffbh123456** [cloudwu/skynet#1488] [Q&A]
https://github.com/cloudwu/lua-cjson
```
local cjson = require("cjson")
local str = '{"a": null}'
local tbl = cjson.decode(str)
local a = tbl['a']
print(type(a)) -- 会输出userdata
```

**Comment by sniper00** [comment]
cjson的问题，建议去对应项目下提issue


发自我的iPhone

在 2021年10月14日，下午10:18，JiaZhi Zhou ***@***.***> 写道：

﻿

https://github.com/cloudwu/lua-cjson

local cjson = require("cjson")
local str = '{"a": null}'
local tbl = cjson.decode(str)
local a = tbl['a']
print(type(a)) -- 会输出userdata

**Reply by ffbh123456** [reply]
lua-cjson项目已经关闭issue功能了，所以才提到这里的

**Comment by sniper00** [comment]
那是它用来表达json null的，lua没有对应的设施

发自我的iPhone

在 2021年10月14日，下午10:26，JiaZhi Zhou ***@***.***> 写道：

﻿

lua-cjson项目已经关闭issue功能了，所以才提到这里的

**Reply by ffbh123456** [reply]
https://github.com/rxi/json.lua
这个json会将null值decode成nil
